src/inatinqperf/adaptors/weaviate_adaptor.py

```
@@ -1,257 +1,340 @@
-            msg = "Query vectors must be 1-D with correct dimensionality"
+        if vector.ndim != 1 or vector.shape[0] != self.dim:
+            msg = "Query vector must be 1-D with dimensionality matching the index"
             raise ValueError(msg)
 
-        vector_json = json.dumps(query_vector.tolist())
-        query_str = (
-            "{\n  Get {\n    "
-            f"{self.class_name}(nearVector: {{ vector: {vector_json} }}, limit: {topk}) "
-            "{\n      originalId\n      _additional { id distance }\n    }\n  }\n}"
+        builder = (
+            self._client.query.get(self.class_name, [self._ORIGINAL_ID_FIELD])
+            .with_near_vector({"vector": vector.tolist()})
+            .with_limit(topk)
+            .with_additional(["id", "distance"])
         )
-        payload = {"query": query_str}
-        response = self._session.post(self._graphql_endpoint, json=payload, timeout=self.timeout)
 
-        if response.status_code != HTTPStatus.OK:
-            self._raise_error("search request failed", response)
+        if filters:
+            # Weaviate's where clause enables server-side filtering.
+            builder = builder.with_where(filters)
 
-        data = response.json()
-        results = self._extract_results(data)
-
-        search_results = []
-
-        for result in results[:topk]:
-            additional = result.get("_additional", {})
-            distance = float(additional.get("distance", "inf"))
-            id_str = additional.get("id")
-            original = result.get("originalId")
-            if original is not None:
-                data_id = int(original)
-
-            else:
-                try:
-                    data_id = self._validate_uuid(id_str) if id_str is not None else -1
-                except ValueError:
-                    data_id = -1
-
-            search_results.append(SearchResult(id=data_id, score=distance))
+        try:
+            data = builder.do()
```

Shouldn't the client throw an exception? We should raise that instead of silently failing.

---

Again defensive programming. Just because one search query fails for some environmental reason. The benchmark should not halt instead continue to next query and that should be apparent in either qualitative output / performance.